pkg/reconciler/taskrun/resources/taskref.go

```diff
@@ -282,31 +296,48 @@ func readRuntimeObjectAsTask(ctx context.Context, namespace string, obj runtime.
 		vr := trustedresources.VerifyResource(ctx, obj, k8s, refSource, verificationPolicies)
 		// Issue a dry-run request to create the remote Task, so that it can undergo validation from validating admission webhooks
 		// without actually creating the Task on the cluster.
-		if err := apiserver.DryRunValidate(ctx, namespace, obj, tekton); err != nil {
+		o, err := apiserver.DryRunValidate(ctx, namespace, obj, tekton)
+		if err != nil {
 			return nil, nil, err
 		}
-		t := &v1.Task{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "Task",
-				APIVersion: "tekton.dev/v1",
-			},
-		}
-		if err := obj.ConvertTo(ctx, t); err != nil {
-			return nil, nil, fmt.Errorf("failed to convert obj %s into Pipeline", obj.GetObjectKind().GroupVersionKind().String())
+		switch mutatedTask := o.(type) {
+		case *v1beta1.Task:
+			t := &v1.Task{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Task",
+					APIVersion: "tekton.dev/v1",
+				},
+			}
+			mutatedTask.ObjectMeta = obj.ObjectMeta
+			if err := mutatedTask.ConvertTo(ctx, t); err != nil {
+				return nil, nil, fmt.Errorf("failed to convert obj %s into Pipeline", mutatedTask.GetObjectKind().GroupVersionKind().String())
+			}
+			return t, &vr, nil
+		default:
+			return nil, nil, errors.New("mutated object is no longer a v1beta1.Task")
 		}
-		return t, &vr, nil
 	case *v1beta1.ClusterTask:
 		obj.SetDefaults(ctx)
 		// Cleanup object from things we don't care about
 		// FIXME: extract this in a function
 		obj.ObjectMeta.OwnerReferences = nil
 		t, err := convertClusterTaskToTask(ctx, *obj)
+		if err != nil {
+			return nil, nil, err
+		}
 		// Issue a dry-run request to create the remote Task, so that it can undergo validation from validating admission webhooks
 		// without actually creating the Task on the cluster
-		if err := apiserver.DryRunValidate(ctx, namespace, t, tekton); err != nil {
+		o, err := apiserver.DryRunValidate(ctx, namespace, t, tekton)
+		if err != nil {
 			return nil, nil, err
 		}
-		return t, nil, err
+		switch mutatedTask := o.(type) {
+		case *v1.Task:
```

Comment:
should this `case` be checking `v1beta1.Task` or `v1.Task`?

Reply:
This should be a `v1.Task` since the function call in line 318 which converts clusterTask to Task returns a `v1.Task`.